src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -571,10 +577,214 @@ public void testMinBaseQualityScore() throws Exception {
 
     }
 
+    @DataProvider
+    public Object[][] getContaminationCorrectionTestData() {
+        // This bam is a snippet of GATKBaseTest.NA12878_20_21_WGS_bam artificially contaminated
+        // at 15% with reads from another sample. We use such a high contamination fraction to ensure
+        // that calls will be noticeably different vs. running without -contamination at all.
+        final String contaminatedBam15Percent = largeFileTestDir + "contaminated_bams/CEUTrio.HiSeq.WGS.b37.NA12878.CONTAMINATED.WITH.HCC1143.NORMALS.AT.15PERCENT.bam";
+
+        final SimpleInterval traversalInterval = new SimpleInterval("20", 10100000, 10150000);
+
+        // File specifying the per-sample contamination for contaminatedBam15Percent
+        final String contaminationFile = TEST_FILES_DIR + "contamination_file_for_CEUTrio.HiSeq.WGS.b37.NA12878.CONTAMINATED.WITH.HCC1143.NORMALS.AT.15PERCENT";
```

Comment:
This file only lists the contamination for one bam. What happens if there are many fields in the contamination file? or if they don't match with the samples the tool sees?  It seems the there should be tests for these cases, and since the contamination logic seems to be in obfuscated within the haplotype caller and its not tested anywhere.

Reply:
There do seem to be tests that amount to showing the file doesn't throw, I think there should be some tests somewhere for the edge cases in these files.

Reply:
There are direct unit tests in `AlleleBiasedDownsamplingUtilsUnitTest` covering loading contamination files with multiple samples, as well as tests in `StandardCallerArgumentCollectionUnitTest` for the multi-sample case.